### DISCUSSION TITLE
Sistema de Autenticación

### DISCUSSION THREAD

**Discussion by parejajd** [casanaredevs/portal#29] [Q&A]
Saludos,

No soy experto en angular, por eso creo este espacio para que conceptualicemos que herramienta para Auth vamos a usar.

¿qué opinan de esta https://github.com/auth0/auth0-angular?

**Comment by guillermoplus** [comment]
Que dice @aldairguarupe sobre este módulo? Este servicio de auth0 es Open source?

**Reply by aldairguarupe** [reply]
@guillermoplus es toda suya la tarea si queda algo pendiente me dice y yo le colaboro, aunque no creo que me deje algo

**Comment by guillermoplus** [comment]
Jajaja no pues gracias @aldairguarupe me gusta el apoyo de corazón amigo jaja

**Reply by parejajd** [reply]
Auth0 es un SaaS pago, que tiene capa gratuita, si deciden irnos por esa, hablen ahora!

**Reply by torreswil** [reply]
Pues dice que alcanza para 7000 usuarios, yo creo que el día que lleguemos a eso ya tenemos para pagar o para cambiar de método. Parece buena idea. (Si fuera en Laravel lo haría con Pasaport + Socialite gratis forever no sé si allá algo similar en .net)

**Reply by joredher** [reply]
Pues, considero que sería mejor manejar toda la autenticación desde el Back. No entiendo mucho como funciona la de Angular. 

Opino lo mismo que @torreswil  se podría manejar algo así en .net?

Nota: La responsabilidad iría mejor en el Back. Lo que haría el Front es solo manejar esos estados.

**Reply by parejajd** [reply]
Si, yo he usado Auth0 y las implementaciones propias de .NET para el manejo de autenticación, en cualquier caso, siempre requieren que el backend lo contenga o lo referencie para el tema de los tokens JWT

Entonces listo, le hacemos así.

Yo monto el API sencillo y dejo montado una implementación de autenticación con GitHub y tokens JWT y apenas lo tenga coordinamos como sería la puesta de ese mecanismo en el front.

Tomaré la estructura propuesta por @torreswil en el chat

Nombre
Descripción
Modalidad: Virtual, Presencial, Mixto ...
Tipo: Kata, Workshop, Encuentro Deportivo ...
Lugar
Fecha y hora
Speakers
Requerimientos

**Reply by torreswil** [reply]
Me faltó el campo para guardar la URL de la imagen del evento, no sé si vamos a permitir cargar imágenes o solo linkear.
